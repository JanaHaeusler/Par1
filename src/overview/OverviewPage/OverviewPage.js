import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import GameList from '../GameList'

OverviewPage.propTypes = {
  savedGameProfiles: PropTypes.object.isRequired,
  deleteGameProfile: PropTypes.func.isRequired,
  prepareEditPage: PropTypes.func.isRequired,
  prepareDetailsPage: PropTypes.func.isRequired,
}

export default function OverviewPage({
  savedGameProfiles,
  deleteGameProfile,
  prepareEditPage,
  prepareDetailsPage,
}) {
  return (
    <>
      <Headline>Your Games</Headline>
      <GameList
        savedGameProfiles={savedGameProfiles}
        onDelete={deleteGameProfile}
        onEdit={prepareEditPage}
        onDetails={prepareDetailsPage}
      />
    </>
  )
}

const Headline = styled.h1`
  margin: 0 0 15px 0;
  text-align: center;
  text-transform: uppercase;
  color: var(--light);

  &::after {
    margin: 8px auto 30px;
    display: block;
    width: 120px;
    height: 2px;
    content: '';
    background-color: var(--light);
    align-content: center;
  }
`
