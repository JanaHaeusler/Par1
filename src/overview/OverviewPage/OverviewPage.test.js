import { render } from '@testing-library/react'
import OverviewPage from './OverviewPage'

const testProps = {
    savedGameProfiles: {
        byId: {
          1: {
            location: 'Horner Racecourse',
            date: '2020-12-24',
            players: 'John, Jane',
            winner: 'Jane',
            shots: '89',
            _id: '1',
          },
          2: {
            location: 'City Park',
            date: '2020-01-01',
            players: 'Fritz, Hanni, Nanni',
            winner: 'Fritz',
            shots: '35',
            _id: '2',
          },
        },
        allIds: [1, 2],
      },
    deleteGameProfile: jest.fn(),
    prepareEditModus: jest.fn(),
    showCreatePage: jest.fn(),
  }

describe('OverviewPage', () => {
    it('renders correctly', () => {
        const { container } = render(<OverviewPage {...testProps}/>)
        expect(container.firstChild).toMatchSnapshot()
    })
})