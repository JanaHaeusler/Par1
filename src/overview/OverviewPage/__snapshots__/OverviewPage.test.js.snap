// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewPage renders correctly 1`] = `
.c0 {
  margin: 0 0 15px 0;
  text-align: center;
  text-transform: uppercase;
  color: var(--text-light);
}

.c0::after {
  margin: 8px auto 30px;
  display: block;
  width: 120px;
  height: 2px;
  content: '';
  background-color: var(--text-light);
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

<h1
  class="c0"
>
  Your Games
</h1>
`;
